docs/sphinx/examples/cpp/other/builder/builder.cpp

```diff
@@ -114,6 +122,37 @@ int main() {
     counts.dump();
   }
 
+  {
+    // In a simulated environment, it is sometimes useful to be able to
+    // specify an initial state vector. The initial state vector is 2 to the
+    // power `n` where `n` is the number of qubits.
+
+    // In this example, we create a kernel template `sim_kernel` that captures
+    // the variable `init_state` by reference.
+    auto sim_builder = cudaq::make_kernel();
+    std::vector<std::complex<double>> init_state;
```

Comment:
we should use `std::vector<cudaq::simulation_scalar>>` instead so this code will work for any of the simulators (wrt their floating point precision)